src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,50 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_LENGTH = 10;
+    private int size;
+
+    private class Element<K, V> {
+        private K key;
+        private V value;
+    }
+
+    private Element<K, V>[] storage;
+
+    public StorageImpl() {
+        storage = new Element[ARRAY_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < storage.length; i++) {
+            if (storage[i] == null) {
+                storage[i] = new Element<>();
+                storage[i].key = key;
+                storage[i].value = value;
+                size++;
+                break;
+            } else if (storage[i].key == key || (storage[i].key != null
+                    && storage[i].key.equals(key))) {
```

Comment:
```suggestion
            } else if (storage[i].key == key || storage[i].key != null
                    && storage[i].key.equals(key)) {
```
redundant brackets. read about logical operators' priority